app/code/Magento/InventoryShippingAdminUi/Plugin/Sales/Block/Shipment/BackButtonUrlOnNewShipmentPagePlugin.php

```diff
@@ -7,42 +7,59 @@
 
 namespace Magento\InventoryShippingAdminUi\Plugin\Sales\Block\Shipment;
 
+use Magento\Framework\App\ObjectManager;
+use Magento\InventoryShippingAdminUi\Model\IsOrderSourceManageable;
 use Magento\Shipping\Block\Adminhtml\Create;
 use Magento\InventoryShippingAdminUi\Model\IsWebsiteInMultiSourceMode;
 
+/**
+ * Class BackButtonUrlOnNewShipmentPagePlugin
+ */
 class BackButtonUrlOnNewShipmentPagePlugin
 {
     /**
      * @var IsWebsiteInMultiSourceMode
      */
     private $isWebsiteInMultiSourceMode;
 
+    /**
+     * @var IsOrderSourceManageable
+     */
+    private $isOrderSourceManageable;
+
     /**
      * @param IsWebsiteInMultiSourceMode $isWebsiteInMultiSourceMode
+     * @param IsOrderSourceManageable $isOrderSourceManageable
      */
     public function __construct(
-        IsWebsiteInMultiSourceMode $isWebsiteInMultiSourceMode
+        IsWebsiteInMultiSourceMode $isWebsiteInMultiSourceMode,
+        IsOrderSourceManageable $isOrderSourceManageable = null
     ) {
         $this->isWebsiteInMultiSourceMode = $isWebsiteInMultiSourceMode;
+        $this->isOrderSourceManageable = $isOrderSourceManageable ??
```

Comment:
Plugins are not covered with Backward Compatibility policy, constructor may be updated without additional checks